docs/usage-guide/topics/ch04-software-architecture.md

```diff
@@ -0,0 +1,21 @@
+# Software Architecture
+
+User interaction with s2n-tls happens primarily through the `s2n_connection` and `s2n_config` structures.
```

Comment:
I'm not sure you currently have enough new information for this to be an entirely new chapter. Is there a way you can work this information into the existing chapters? Maybe connection.md because it's one of the first chapters.

Reply:
One other way that I think would work is to combine the software architecture, connection, and config chapters together and name it `Software Architecture`. I think both connection and config are parts of the software architecture. I don't think adding general software architecture documentation to connection.md alone is appropriate. Hence, I would say either we use three chapters to describe software architecture, or we combine all three chapters into one. I prefer the former, since it makes each chapter to serve one purpose. It also follows a logical flow where we discussed a general concept, and then gets down to the specific about connection and config.